packages/arb-token-bridge-ui/src/components/RecoverFunds.tsx

```diff
@@ -0,0 +1,530 @@
+
+async function createRetryableTicket({
+  inboxAddress,
+  childProvider,
+  signer,
+  destinationAddress,
+  l2CallValue,
+  gasEstimation,
+  signerAddress
+}: {
+  inboxAddress: string
+  childProvider: StaticJsonRpcProvider
+  signer: Signer
+  destinationAddress: string
+  l2CallValue: BigNumber
+  gasEstimation: ParentToChildMessageGasParams
+  signerAddress: string
+}) {
+  const { nativeToken } = await EthBridger.fromProvider(childProvider)
+
+  if (nativeToken) {
+    const inbox = ERC20Inbox__factory.connect(inboxAddress, childProvider)
+    return inbox.connect(signer).unsafeCreateRetryableTicket(
+      destinationAddress, // to
+      l2CallValue, // l2CallValue
+      gasEstimation.maxSubmissionCost, // maxSubmissionCost
+      destinationAddress, // excessFeeRefundAddress
+      destinationAddress, // callValueRefundAddress
+      gasEstimation.gasLimit, // gasLimit
+      gasEstimation.maxFeePerGas, // maxFeePerGas
+      0, // tokenTotalFeeAmount
+      '0x', // data
+      {
+        from: signerAddress
+      }
+    )
+  }
+
+  const inbox = Inbox__factory.connect(inboxAddress, childProvider)
+  return inbox.connect(signer).unsafeCreateRetryableTicket(
+    destinationAddress, // to
+    l2CallValue, // l2CallValue
+    gasEstimation.maxSubmissionCost, // maxSubmissionCost
+    destinationAddress, // excessFeeRefundAddress
+    destinationAddress, // callValueRefundAddress
+    gasEstimation.gasLimit, // gasLimit
+    gasEstimation.maxFeePerGas, // maxFeePerGas
+    '0x', // data
+    {
+      from: signerAddress,
+      value: 0
+    }
+  )
+}
+
+async function prepareTransaction({
+  address,
+  destinationAddress,
+  childProvider,
+  parentProvider,
+  balanceToRecover
+}: {
+  address: string
+  destinationAddress: string
+  childProvider: StaticJsonRpcProvider
+  parentProvider: StaticJsonRpcProvider
+  balanceToRecover: BigNumber
+}) {
+  // We estimate gas usage
+  const parentToChildMessageGasEstimator = new ParentToChildMessageGasEstimator(
+    childProvider
+  )
+  const aliasedAddress = getAliasedAddress(address)
+
+  const gasEstimation = await parentToChildMessageGasEstimator.estimateAll(
+    {
+      from: aliasedAddress,
+      to: destinationAddress,
+      l2CallValue: balanceToRecover,
+      excessFeeRefundAddress: destinationAddress,
+      callValueRefundAddress: destinationAddress,
+      data: '0x'
+    },
+    await getBaseFee(parentProvider),
+    parentProvider
+  )
+
+  // And we send the request through the method unsafeCreateRetryableTicket of the Inbox contract
+  // We need this method because we don't want the contract to check that we are not sending the l2CallValue
+  // in the "value" of the transaction, because we want to use the amount that is already on child chain
+  const l2CallValue = balanceToRecover
+    .sub(gasEstimation.maxSubmissionCost)
+    .sub(gasEstimation.gasLimit.mul(gasEstimation.maxFeePerGas))
+
+  return {
+    l2CallValue,
+    gasEstimation
+  }
+}
```

Comment:
I strongly feel both these functions are good candidates to move in the token-bridge-sdk utils: since they're pretty generic retryable methods, and later can be used in other places too. Also would clean up the code in this file.

Reply:
It's a very specific function. If we ever reuse it we can move it to `token-bridge-sdk`, but for now I would just keep it here